test/client_test.dart

```diff
@@ -63,6 +63,90 @@ void main() {
       });
     });
 
+    group('connection-operational', () {
+      clientTest('on-connection-operational-fires',
+          (ClientTransportConnection client,
+              FrameWriter serverWriter,
+              StreamIterator<Frame> serverReader,
+              Future<Frame> Function() nextFrame) async {
+        var settingsDone = Completer();
+
+        Future serverFun() async {
+          serverWriter.writeSettingsFrame([]);
+          expect(await nextFrame() is SettingsFrame, true);
+          serverWriter.writeSettingsAckFrame();
+          expect(await nextFrame() is SettingsFrame, true);
+
+          settingsDone.complete();
```

Comment:
Could this be two statements above? (it's not necessary to wait until the ACK has arrived, it's enough the client got the settings frame)